doc/adr/0007-api-service.md

```diff
@@ -0,0 +1,106 @@
+- Deployment
+- Role
+- RoleBinding
+- Service
+
+The specific permissions for the Role resource will be:
+
+```
+resources: ["apps.wego.weave.works"]
+verbs: ["get","create","list"]
+
+resources: ["secret"]
+verbs: ["get","create"]
+
+resources: ["kustomizations.kustomize.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmreleases.helm.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmrepositories.source.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+```
+
+These resources will be applied when installing the GitOps engine.
+
+### Enable access to the wego-api from the client-side.
+
+To enable access to the `wego-api` from the client machine, we will use `kubectl` by running this command: `kubectl -n NAMESPACE port-forward deployment/wego-api :8283`. An example of the output would be: 
+```
+Forwarding from 127.0.0.1:63701 -> 8283
+Forwarding from [::1]:63701 -> 8283
+```
+
+The path to access `wego-api` will be `127.0.0.1:63701` 
+
+This process of exposing `wego-api` will be executed for each command that is executed from CLI. With the exception of the `wego ui run` command that will keep `wego-api` exposure execution in the background for the duration of that command session, so that the UI can access the resource.
+
+### Automate the build of docker images from the `wego-api`.
+
+For production environments, it will be needed a docker repository, a token with permissions to push the image, and public access to those images.
+
+When a `wego` release is made, a Github action will be triggered that will make the corresponding release in the wego-api's Github docker repository using the same tag used for the` wego` release.
+
+At runtime, we will use the version LD flag to tell `wego` which tag to use in the `wego-api` deployment.
+
+### Use source to build locally the wego-api docker image for development mode.
+
+Part of the mechanism from the previous point will be used so that it can also be applied in development mode. And thus avoid wasting time publishing / downloading to a docker registry. Currently, there are many tools on the market to achieve this task, to mention a few there are `garden.io` and` tilt`.
+
+The recommendation is to use `garden.io` because they use yaml files to set the configuration which is something the team is already used to. Rather than `tilt` that relies on python which would enforce learning that language. 
+
+### Reuse the local build of the wego-api inside Github actions for testing.
+
+Since creating a pull request in `wego` triggers several tests. It will also be required to use the construction system of the previous point to create an image that is completely directed for the tests.
+
+To ensure that the construction of images of the wego-api in Github actions is as fast as possible, a mechanism will be implemented that allows sharing the context of the docker builder. One option that seems simple to implement is https://github.com/docker/build-push-action.
+
+## Consequences
+
```

Comment:
Please elaborate on current CLI actions that will be updated.  i.e. version should not report the version of the API server.  Does status change to interact with the API?  etc.

Reply:
Initially the only command using `wego-api` will be `wego ui run` I'll fix the ADR to mentions this.

Reply:
> Please elaborate on current CLI actions that will be updated. i.e. version should not report the version of the API server. Does status change to interact with the API? etc.

I meant to say version **should** report the version of the API server.  I realize only the UI will utilize it for now but we will want to inform the user about it and version is the first place.